### PR TITLE
Add and update PodSecurityPolicy to match SecurityContext

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This changeset add and update `PodSecurityPolicy` for open source and enterprise components to match the recent work for SecurityContext in [1]. This is also required for all components to work in a CIS hardened RKE2 cluster.

To summarize the changes:
* Update Elasticsearch and Fluentd capabilities in `PodSecurityPolicy` to match `SecurityContext`.
* Add missing `PodSecurityPolicy` for all enterprise components. Enterprise components like packetcapture, esgateway, l7-log-collector, etc. are able to be deployed without creating `PodSecurityPolicy` manually.
* Manage Prometheus operator `ClusterRole` and `ClusterRoleBinding` in operator.
* use `UsePSP` as the only flag to determine whether we need to render `PodSecurityPolicy` or not.

[1] https://github.com/tigera/operator/pull/2433

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/monitor/monitor_test.go
Comment (rene-dekker) [inline pkg/render/monitor/monitor_test.go:159]: I suppose this needs to be implemented still?

File: pkg/render/packet_capture_api.go
Comment (rene-dekker) [inline pkg/render/packet_capture_api.go:73]: nit: whether or not can be shortened to whether

File: pkg/render/logstorage/esmetrics/elasticsearch_metrics.go
Comment (rene-dekker) [inline pkg/render/logstorage/esmetrics/elasticsearch_metrics.go:102]: should the role and binding be removed if !usePSP?

File: pkg/render/applicationlayer/applicationlayer.go
Comment (rene-dekker) [inline pkg/render/applicationlayer/applicationlayer.go:181]: Should the role and binding be removed if !usePSP?

File: pkg/render/csi.go
Comment (rene-dekker) [inline pkg/render/csi.go:385]: Should we remove some of this stuff if !usePSP?

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:260]: Should we delete the role and binding if !usePSP?

File: pkg/render/monitor/monitor_test.go
Comment (hjiawei) [reply pkg/render/monitor/monitor_test.go:159]: Implemented.

File: pkg/render/packet_capture_api.go
Comment (hjiawei) [reply pkg/render/packet_capture_api.go:73]: This comment is copied from other places. I would like to make it consistent with other places for now.

File: pkg/render/logstorage/esmetrics/elasticsearch_metrics.go
Comment (hjiawei) [reply pkg/render/logstorage/esmetrics/elasticsearch_metrics.go:102]: I had a similar discussion with one of our operator developers. We are not deleting PSP related resources (policy itself, (Cluster)Role, and (Cluster)RoleBinding) when `usePSP` is `false`. Maybe we can address this together in a future PR.

File: pkg/render/logstorage/esgateway/esgateway.go
Comment (rene-dekker) [inline pkg/render/logstorage/esgateway/esgateway.go:84]: // Whether or not
is not a correct phrase in this context, as or not is redundant.

// Whether

**Review by rene-dekker — APPROVED** [review]
lgtm

File: pkg/render/logstorage/esgateway/esgateway.go
Comment (hjiawei) [reply pkg/render/logstorage/esgateway/esgateway.go]: Comments are updated.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
